### ISSUE TITLE
Format

### ISSUE THREAD

**Issue by beslst** [BurksEngineering/TwincatLibraryUtilities#1]
I made a bit sense out of the .library format, the approach with the key values pairs is a bit flawed, cause they are not actually key value pairs, but this is only a string table.

I managed to correctly parse the string table and will infest a bit, when I have time, to check where the strings in the table are actually used - so far I am also assuming that it's key value pairs with some heuristics.

I can do a PR when i figured out the format a bit more and know where the strings are referenced. Given that you want to keep this repo up to date

https://github.com/Zeugwerk/Twinpack/blob/main/TwinpackShared/LibraryPropertyReader.cs

**Comment by BurksEngineering** [comment]
Good sleuthing! I'm not very familiar with different binary or archive formats, I literally just opened up the binary in notepad and started searching for stuff :P

Keep me posted as you discover more about the format.

**Comment by beslst** [comment]
I am literally and figuratively done with the library project information. I think for the dependencies I’ll stay with a regex approach for now as it’s pretty reliable.

https://github.com/Zeugwerk/Twinpack/blob/main/TwinpackShared/LibraryReader.cs

this is one of the weirder formats I had to deal with, especially how the library categories are encoded cost some headache. Information we don’t even need, but since this info is at some arbitrary location in the binary I had to go through this :D

since 4024.53 the information is stored in a normal xml, which makes it much easier.

fyi @seehma

cheers
